Vagrantfile

```diff
@@ -44,6 +44,25 @@ Vagrant.configure("2") do |config|
     if host_arch == "x86_64"
       lv.machine_type = "q35"
     end
+    # The VM needs one additional virtio-net device and iommu enabled
+    # for the vfio tests.
+    if job == "VFIO" and host_arch == "x86_64"
+      # Overwrite the default -machine already built by vagrant as
+      # additional arguments aren't allowed on `machine_type` property.
+      lv.qemuargs :value => "-machine"
+      lv.qemuargs :value => "q35,accel=kvm,kernel-irqchip=split,usb=off,dump-guest-core=off"
+      lv.qemuargs :value => "-device"
+      lv.qemuargs :value => "intel-iommu,intremap=on,caching-mode=on,device-iotlb=on"
+      # Currently the vfio test picks the last virtio-net device from lspci's
+      # output. Here we add the device in a PCIe root port with higher slot
+      # number on the hope it will be the last in the list.
+      lv.qemuargs :value => "-device"
+      lv.qemuargs :value => "pcie-root-port,port=0x16,chassis=7,id=pcie.7,multifunction=on,bus=pcie.0,addr=0x04"
```

Comment:
Any reason to use 0x04, rather than a really high slot number?  I think q35 has integrated devices in slot 0x1f and maybe 0x1e as well, but you can still go much higher than 4.

Reply:
No particular reason. I can change it to a higher slot so decrease the probability of the test not catching that device.